cypress/integration/courses.spec.js

```diff
@@ -0,0 +1,74 @@
+describe('courses page', () => {
+  before(() => {
+    cy.loginAndVisit();
+  });
+
+  it('navigates to Courses page from Summary', () => {
+    cy.contains('Courses').click();
+    cy.location().should(loc => {
+      expect(loc.pathname).to.contain('courses');
+    });
+  });
+
+  it('contains a courses table', () => {
+    cy.get('table[course-table]');
+  });
+
+  it('removes ECS courses from 2018-19', () => {
+    cy.get('[data-event-type=selectAllCourseRows].checkbox-container').click();
+    cy.get('span.tool-icon.glyphicon.glyphicon-trash').click();
+    cy.get('.delete-course-modal-footer').within($modal => {
+      cy.server();
+      cy.route('PUT', '**/courseView/**').as('deleteCourses');
+      cy.get('button.btn-danger').click();
+    });
+    cy.wait('@deleteCourses');
+  });
+
+  it('adds ECS courses from 2015-16', () => {
+    cy.get('.glyphicon-plus').click();
+    cy.contains('Add Multiple Courses');
+
+    cy
+      .contains('Source')
+      .next('.col-sm-4')
```

Comment:
.col-sm-4 seemed dangerous as a hook, as it is a very generic bootstrap class used for layout, and would be  subject to change.

I spent a little time digging around and found this in the Cypress docs.
https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements

<img width="778" alt="screen shot 2018-06-14 at 2 24 46 pm" src="https://user-images.githubusercontent.com/1776859/41439207-befb23c6-6fde-11e8-9d23-584b91edc473.png">

Their recommended solution using data attributes looks good to me, and in the future we can start sprinkling in these cypress hooks around our code as needed for tests.

For now, trying to pick hooks that are as stable (further down on the table) as possible seems sensible.

I commented on some of the selectors below, but really all the selectors used should be considered to see if we can make them more stable (until we're ready to migrate fully to data attribute hooks).